### PR TITLE
EOS-14372: Bucket list failing to return list for 500 bucket account

### PR DESCRIPTION
**Pull request by soniyamoholkar**
# Backend

## Problem Statement
<pre>
Story Ref (if any): EOS-14372
</pre>
## Unit testing on RPM done
<pre>
Yes
</pre>
## Problem Description 
<pre>
Bucket list failing to return list for 500 bucket account
</pre>
## Solution: 
<pre>
Stored network configuration fetched from provisioner in in-memory. So while creating s3_url, it will get the cluster IP from in-memory instead of fetching it from provisioner evry time. 
</pre>
## Unit Test Cases
<pre>

</pre>
Signed-off-by: Soniya Moholkar

### CODE REVIEW

File: csm/plugins/cortx/provisioner.py [hunk 1]
```diff
@@ -251,23 +252,27 @@ async def get_network_configuration(self):
...
-                        mgmt_vip=params[self.PRVSNR_NETWORK_PARAM_VIP],
-                        cluster_ip=params[self.PRVSNR_NETWORK_PARAM_CIP]
-                    )
-            except Exception as error:
-                Log.error(f"Provisioner api error : {error}")
-                raise NetworkConfigFetchError(f"Network configuration fetching failed: {error}")
-
-        return await self._await_nonasync(_command_handler)
+        if not self.network_config:
+            Log.debug("Netword config is not present in in-memory.")
```
Comment (kupranay) [inline csm/plugins/cortx/provisioner.py:256]: Please correct spelling of Networkd to network

File: csm/plugins/cortx/provisioner.py [hunk 2]
```diff
@@ -58,6 +58,7 @@ class ProvisionerPlugin:
     PRVSNR_NETWORK_PARAM_CIP = 'network/cluster_ip'
 
     def __init__(self, username=None, password=None):
+        self.network_config = None
```
Comment (kupranay) [inline csm/plugins/cortx/provisioner.py:61]: With this process on every class initialization will set network_config to none. Lets network config to in -memory conf file

**Review by kupranay — APPROVED** [review]

File: csm/plugins/cortx/provisioner.py (hunk 1 above)
Comment (soniyamoholkar) [reply csm/plugins/cortx/provisioner.py]: Done

File: csm/plugins/cortx/provisioner.py (hunk 2 above)
Comment (soniyamoholkar) [reply csm/plugins/cortx/provisioner.py]: Saved network config in Conf.

**Review by kupranay — APPROVED** [review]

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**